swan-lake/learn-the-platform/ballerina-persist/persist-client-api.md

```diff
@@ -0,0 +1,245 @@
+---
+layout: ballerina-persist-documentation-left-nav-pages-swanlake
+title: Persist Client API
+description: The sections gives details on the generated Persist Client API.
+keywords: ballerina, programming language, ballerina packages, persist, persist client, persist client api
+permalink: /learn/ballerina-persist/persist-client-api/
+active: persist_client_api
+intro: Persist client is used to perform CRUD operations on the data source. Each entity type will have five resource methods for each operation. The resource methods are get, get(get by identity), post, put, and delete. Persist CLI tool will generate the derived Entity Types and the Clients from the model definition.
+redirect_from:
+- /learn/ballerina-persist/persist-client-api
+---
+
+## Derived Entity Types
+
+Ballerina record types used in the generated clients are derived from the entity types defined in the data model.
+
+Example of a defined entity,
+```ballerina
+type Workspace record {|
+    readonly string id;
+    string workspaceType;
+    Building location; // 1-n relation (parent)
+    Employee? employees; // 1-1 relation (child)
+|};
+```
+
+There are six types of derived entity types:
+1. Entity Types  
+    These are the types defined in the data model. These are used to indicate the data structure in the data source. The entity who is the association parent will include the foreign key field(s).
+    ```ballerina
+    public type Workspace record {|
+        readonly string id;
+        string workspaceType;
+        string locationCode;
+    |};
+    ```
+    
+2. Insert Types
+    These are the records used to insert data in the data source. This is same as the Entity Type.
+    ```ballerina
+    public type WorkspaceInsert Workspace;
+    ```
+
+3. Update Types
+    These are the records used to update data in the data source. These are entity types without the identity fields. All fields will be optional. Only the fields for which values are provided will be updated.
+    ```ballerina
+    public type WorkspaceUpdate record {|
+        string workspaceType?;
+        string locationCode?;
+    |};
+    ```
+   
+4. Optionalized Types
+    These are the same as the Entity Types, but all the fields are made optional. This type is not directly used in any operations.
+    ```ballerina
+    public type WorkspaceOptionalized record {|
+        readonly string id?;
+        string workspaceType?;
+        string locationCode?;
+    |};
+    ```
+   
+5. With Relations Types
+    This types inherits all fields from the corresponding `Optionalized` type, and adds the relation fields as optional. This type is not directly used in any operations.
```

Comment:
```suggestion
    These types inherit all fields from the corresponding `optionalized` type and add the relation fields as optional. They are not directly used in any operations.
```
Should be either `relational fields` or `related fields`.

Reply:
Let's keep as `relation` fields